gitpod-io/gitpod
# Allow teams to sign up for Usage-Based Billing via Stripe

## Description


Allow teams to sign up for Usage-Based Billing via Stripe.

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/10326

## How to test


1. Join the "Gitpod" team: [invite link](https://jx-stripe-service.preview.gitpod-dev.com/teams/join?inviteId=9d9cd3f1-bad9-4725-8e06-469ce0d955c1) (this enables the Usage-Based feature flag for you)
2. Create a new, different team ✨ 
3. Navigate to your Team Settings > Team Billing
4. Wait for the Usage-Based UI to appear above the Team Plans
5. Upgrade your team to usage-based by providing a fake credit card (e.g. 4242 4242 4242 4242, 4/24, 424)
6. This should successfully create a Stripe Customer for your team, visible in our [Stripe Dashboard](https://dashboard.stripe.com/test/customers) (credentials in 1Password)

| No Team Billing | Add Credit Card | Active Team Billing | Stripe Dashboard |
| --- | --- | --- | --- |
| <img width="1437" alt="Screenshot 2022-06-01 at 16 54 30" src="https://user-images.githubusercontent.com/599268/171435393-37886683-09bb-4b33-8d7e-da04673c5212.png"> | <img width="1200" alt="Screenshot 2022-06-01 at 16 55 02" src="https://user-images.githubusercontent.com/599268/171435405-f1285e38-956d-40df-bdc2-0a12d54d6772.png"> | <img width="1438" alt="Screenshot 2022-06-01 at 16 58 35" src="https://user-images.githubusercontent.com/599268/171435669-de8f677b-f448-49eb-97be-eff9b7ca499c.png"> | <img width="1242" alt="Screenshot 2022-06-01 at 16 56 42" src="https://user-images.githubusercontent.com/599268/171435413-bba639b9-b4d3-43bf-b220-afb01d06c07d.png"> |


## Release Notes

```release-note
NONE
```

## Documentation
